Integrations/integration-Tufin.yml

```diff
@@ -0,0 +1,321 @@
+  SDN, and public cloud to identify vulnerable access paths of an attack
+detaileddescription: Learn more about Tufin at [https://www.tufin.com/](https://www.tufin.com/)
+configuration:
+- display: SecureTrack User Credentials
+  name: SecureTrack-User
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: SecureTrack IP or FQDN
+  name: SecureTrack-Server
+  defaultvalue: 127.0.0.1
+  type: 0
+  required: true
+- display: Do not verify SSL Cert
+  name: unsecure
+  defaultvalue: "true"
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |
+    import json
+    import requests
+
+    # Disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+    # remove proxy if not set to true in params
+    if not demisto.params().get('proxy'):
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+
+    # Request something from TOS
+    def tos_request(path, params=None, headers=None):
+        """ Function to access TOS via REST API """
+        if headers is None:
+            headers = {
+                'accept': "application/json",
+                'content-type': "application/json",
+                'cache-control': "no-cache"
+            }
+
+        # Get Configuration
+        st_ip = demisto.params()['SecureTrack-Server']
+        st_user = demisto.params()['SecureTrack-User']['identifier']
+        st_pass = demisto.params()['SecureTrack-User']['password']
+        verify_ssl = not demisto.params().get('unsecure', False)
+        url = 'https://' + st_ip + path
+
+        # Go do
+        try:
+            res = requests.get(url, params=params, headers=headers, auth=(st_user, st_pass), verify=verify_ssl)
+        except requests.exceptions.RequestException as e:
+            return_error(str(e))
+
+        # Check output
+        if res.status_code != requests.codes.ok:
+            if res.status_code == 401:
+                return_error('TOS Reached, Auth Failed. Please check your credentials')
+            else:
+                return_error('Error {} Reaching {} to TOS: {}'.format(res.status_code, res.url, res.reason))
+        else:
+            try:
+                return res.json()
+            except:
+                return res.content
+
+
+    def path_finder():
+        """ Sample query: querystring = {"src":"10.80.80.0","dst":"172.16.200.80","service":"tcp:22","includeIncompletePaths":"true"} """
+
+        # Build the query from user input
+        querystring = {
+            "src": demisto.args()['Source'],
+            "dst": demisto.args()['Destination'],
+            "service": demisto.args().get('Service', 'Any'),
+            "includeIncompletePaths": "true"
+        }
+
+        # Define the basic output for the function, augmenting later with TOS data
+        entry = {
```

Comment:
Please build the entry once when you return it, instead of first initialize it and then return it.

Reply:
This would create duplicate definitions of the entry var, increasing future modification time.